crates/sparrow-compiler/src/types/inference.rs

```diff
@@ -295,6 +295,21 @@ fn solve_type_class(
                 })
                 .collect();
 
+            // TODO: Map type arguments are reported at the location of the outer `map`,
+            // which is confusing.
+            // The actual error is that `large_string` is the `map.key` type,
+            // and we're expected `string`.
+            //
+            // error[E0010]: Invalid argument type(s)
+            //   --> Query:1:10
+            //   |
+            // 1 | { value: get("test" , Input.readings )}
+            //   |          ^^^ ------   -------------- Type: large_string
+            //   |          |   |
+            //   |          |   Type: string
+            //   |          Invalid types for call to 'get'
+            //   |
+            //   = Expected 'key'
```

Comment:
This is weird -- unsure if there's a quick ux improvement for this. the current errors make it difficult since they have a formatted string inserting these types.

Reply:
I'm not sure I follow, and I'm not sure that's the error. I think this is actually saying that `large_string` isn't valid as a `key` (eg., the type class) since we can't hash large string.

Reply:
we can hash large string though..I think it's something to do with the literal key input was a `string` and the `map.key` was a `large_string`.

Reply:
Do you have logic for least upper bound? Specifically, the least upper bound of `string` and `large_string` should be `large_string`.

Reply:
Ah, that was part of the problem -- that and comparing the incorrect types for doing promotion